lib/spack/spack/test/cmd/ci.py

```diff
@@ -123,10 +124,14 @@ def test_specs_staging(config):
 
 
 def test_ci_generate_with_env(tmpdir, mutable_mock_env_path, env_deactivate,
-                              install_mockery, mock_packages, project_dir_env):
+                              install_mockery, mock_packages, env_vars):
     """Make sure we can get a .gitlab-ci.yml from an environment file
        which has the gitlab-ci, cdash, and mirrors sections."""
-    project_dir_env(tmpdir.strpath)
+    env_vars({
+        'CI_PROJECT_DIR': tmpdir.strpath,
+        'CI_JOB_NAME': 'generate-pipeline',
+        'CI_PIPELINE_ID': '42'
+    })
```

Comment:
wouldn't it be easier to do this by using the existing `working_env` fixture, and then just updating the environment in the test as needed? (`os.environ.update({...})`)?

If that's not easier, can you refactor the fixture to uses the `working_env` fixture instead of doing your own environment tracking?

Reply:
Yeah, let me look at that, I was unaware of it.